### ISSUE TITLE
brew install subversion --with-java failed to build on 10.12.6

### ISSUE THREAD

**Issue by PSFC** [Homebrew/homebrew-core#33856]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
Some output from one xml library I built remained but should not interfere with brew.
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Build subversion 1.9.x in order to obtain the JavaHL libraries for Subclipse plug-in for Eclipse.
- What happened (include command output)
The build failed.

Log: https://gist.github.com/905773112a0a918610c3d85b8e8e5f8e

Output:

```
brew install subversion --with-java
==> Downloading https://www.apache.org/dyn/closer.cgi?path=subversion/subversion-1.10.2.tar.bz2
Already downloaded: /Users/ithran_einhorn/Library/Caches/Homebrew/downloads/891bd5f606e2553db3534719644e0fee38f62a2e8f0d048d11bbb3092e982904--subversion-1.10.2.tar.bz2
==> Patching
patching file subversion/bindings/swig/perl/native/Makefile.PL.in
patching file build/get-py-info.py
==> Downloading https://www.apache.org/dyn/closer.cgi?path=serf/serf-1.3.9.tar.bz2
Already downloaded: /Users/ithran_einhorn/Library/Caches/Homebrew/downloads/3f4e2cb0f268fb9db7a761edccb60bdbdacb9923f8b2140e8c0a30b6792958a6--serf-1.3.9.tar.bz2
==> /usr/local/opt/scons/bin/scons PREFIX=/usr/local/Cellar/subversion/1.10.2/libexec/serf GSSAPI=/us
==> /usr/local/opt/scons/bin/scons install
==> ./configure --prefix=/usr/local/Cellar/subversion/1.10.2 --enable-optimize --disable-mod-activati
==> make
==> make install
==> make tools
==> make install-tools
==> make swig-py
==> make install-swig-py
==> make swig-pl
==> make install-swig-pl
==> make javahl
==> make install-javahl
==> make swig-rb EXTRA_SWIG_LDFLAGS=-L/usr/lib
Last 15 lines from /Users/ithran_einhorn/Library/Logs/Homebrew/subversion/14.make:
2018-11-06 15:56:29 -0500

make
swig-rb
EXTRA_SWIG_LDFLAGS=-L/usr/lib

/bin/sh "/private/tmp/subversion-20181106-21065-7jbhe3/subversion-1.10.2/libtool" --tag=CC --silent --mode=compile none  -I/usr/local/opt/sqlite/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl/include -F/usr/local/Frameworks  -DDARWIN -DSIGPROCMASK_SETS_THREAD_MASK -DDARWIN_10  -I/private/tmp/subversion-20181106-21065-7jbhe3/subversion-1.10.2/subversion/bindings/swig/ruby/libsvn_swig_ruby -prefer-pic -c -o subversion/bindings/swig/ruby/svn_client.lo subversion/bindings/swig/ruby/svn_client.c
/private/tmp/subversion-20181106-21065-7jbhe3/subversion-1.10.2/libtool: line 1760: none: command not found
make: *** [subversion/bindings/swig/ruby/svn_client.lo] Error 1

READ THIS: https://docs.brew.sh/Troubleshooting

These open issues may also help:
Subversion 1.11.0 https://github.com/Homebrew/homebrew-core/pull/33530
subversion: enable i18n support https://github.com/Homebrew/homebrew-core/pull/33162
```

- What you expected to happen
A successful build of subversion.

- Step-by-step reproduction instructions (by running `brew install` commands)
brew install subversion --with-java

**Comment by fxcoudert** [comment]
Possibly a duplicate of https://github.com/Homebrew/homebrew-core/issues/33250
Can you run `clang --version` and report the output? It is possibly that you need to accept the Xcode licence with `sudo xcodebuild -license`.

**Comment by PSFC** [comment]
I did that and got satisfactory output:

```
clang --version
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Comment by vickllny** [comment]
@fxcoudert Good morning.
At first, I installed javahl on xcode10.1. This problem occurred.
The installation process gave me some error messages that are different from yours. 
I probably know from the error message that the version of sdk is incorrect, the sdk version is 10.14.
So. I am trying to install a lower version of xcode10.1. It prompts you the same as your error message.
Then I uninstalled Xcode and Clt (Command line tools), installed an earlier version of Xcode and Clt, the version number is 9.4.1, and the miracle happened. It installed successful
